pkg/virt-controller/services/serialconsolelog.go

```diff
@@ -0,0 +1,91 @@
+package services
+
+import (
+	"fmt"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/util"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+func generateSerialConsoleLogContainer(vmi *v1.VirtualMachineInstance, image string, config *virtconfig.ClusterConfig) *k8sv1.Container {
+	// TODO: eventually add a cluster-wide default
+	if (vmi.Spec.Domain.Devices.AutoattachSerialConsole == nil || *vmi.Spec.Domain.Devices.AutoattachSerialConsole == true) && vmi.Spec.Domain.Devices.LogSerialConsole != nil && *vmi.Spec.Domain.Devices.LogSerialConsole == true {
+		var serialPort uint = 0
+
+		followretry := "-F"
+		quiet := "--quiet"
+		nodup := "-n+1"
+		logFile := fmt.Sprintf("%s/%s/virt-serial%d-log", util.VirtPrivateDir, vmi.ObjectMeta.UID, serialPort)
+		args := []string{quiet, nodup, followretry, logFile}
+
+		resources := resourcesForSerialConsoleLogContainer(false, false, config)
+		noPrivilegeEscalation := false
```

Comment:
Make these const or use `pointer` pkg and inline them instead?

Reply:
done